lib/honeycomb-rails/extensions/action_controller.rb

```diff
@@ -25,6 +25,22 @@ def honeycomb_initialize
         # @return [Hash<String=>Any>]
         attr_reader :honeycomb_metadata
       end
+
+      module InstanceCaptureExceptionsFilters
+        def self.included(controller_class)
+          controller_class.around_action do |base, block|
```

Comment:
A few code structure things:

 * This would more naturally live in `Overrides::ActionControllerInstrumentation`. The idea is that `Extensions` is for adding new methods to the ActionController API that user code is expected to call (as this file does by adding `honeycomb_metadata`) and `Overrides` is for overriding existing behaviour transparently to user code (as we do here). This is a somewhat common separation and naming scheme in the Rails world.
     * `Overrides::ActionControllerInstrumentation` doesn't have an `included` hook yet but no reason it shouldn't. We could still retain the option to turn it off in config, e.g. by checking config in the `included` hook... give or take initialization order fun which would need testing.
 * Rather than the block form of `around_action`, could we define a `honeycomb_attach_exception_metadata` instance method (e.g. in `Overrides::ActionControllerInstrumentation`) and then call `controller_class.around_action :honeycomb_attach_exception_metadata`? That would make testing a lot easier, and also be marginally more idiomatic.

Reply:
🙏

> `Overrides::ActionControllerInstrumentation` doesn't have an included hook yet but no reason it shouldn't.

Can I still keep it inside another module inside `Overrides::ActionControllerInstrumentation`? the `Railtie` `initializer` call doesn't have `HoneycombRails.config` values set yet, and the `HoneycombRails.config.capture_exceptions` -> `...include` really needs to happen in `after_initialize`. Having it in a separate module allows for finer-grained inclusion (not a thing that I feel strongly about, just... the only way I managed to get it working. 😐

Thumbs up to all of the above, thanks for banging against this with me!

Reply:
D'oh, I had a feeling the initialization order might mess things up. Yup sounds great.